[13.0][MIG] l10n_es_account_bank_statement_import_n43: Migration to v.13

ValentinVinagre opened the pull request:
Standard migration. Eliminados los commits administrativos.
Superseed: https://github.com/OCA/l10n-spain/pull/1279

pedrobaeza committed:
[ADD] l10n_es_account_bank_statement_import_n43: Sustituye a l10n_es_bank_statement para importar extractos bancarios N43

Roodin committed:
[IMP] l10n_es_account_bank_statement_import_n43. Importar extractos con monedas distintas de euros segun issue #217 y basado en comentario de @fgarciag2001

omar7r committed:
[IMP] l10n_es_account_bank_statement_import_n43: Se añaden dos comprobaciones extra para obtener el partner extraídas del n43 de Bankia Y Sabadell

pedrobaeza committed:
[FIX] l10n_es_account_bank_statement_import_n43: Populate correctly partner_id

pedrobaeza committed:
[MIG] l10n_es_account_bank_statement_import_n43: Migration to 9.0

Improvements
------------

* Counterpart account automatically detected from N43 type (deactivated for now)
* Tests

pedrobaeza committed:
[IMP] l10n_es_account_bank_statement_import_n43: Tests

pedrobaeza committed:
[FIX] l10n_es_account_bank_statement_import_n43: Fix 'conceptos' default value

pedrobaeza committed:
[FIX] l10n_es_account_bank_statement_import_n43: Avoid error when having other statement formats

pedrobaeza committed:
[MIG] l10n_es_account_bank_statement_import_n43: Migration to 10.0

pedrobaeza committed:
[IMP] l10n_es_account_bank_statement_import_n43: Allow to select between 'fecha_valor' and 'fecha_oper'.

rlizana committed:
[FIX] l10n_es_account_bank_statement_import_n43: al parsear los registros 22, nº documento incorrecto

pedrobaeza committed:
[MIG] l10n_es_account_bank_statement_import_n43: Migration to 11.0

* Metafiles updated
* Test adapted to Python 3
* Test coverage improved

pedrobaeza committed:
[FIX] l10n_es_account_bank_statement_import_n43: Fix tests due to upstream change

pedrobaeza committed:
[IMP] l10n_es_account_bank_statement_import_n43: Don't require n43_date_type

With this, this is compatible with tests that creates journals before this
module is loaded.

pedrobaeza committed:
[MIG] l10n_es_bank_statement_import_n43: Migration to 12.0

* Standard procedure
* Override method for adding the supported format
* README by fragments
* Tests adapted for new date format and order

Commit:
[FIX] l10n_es_account_bank_statement_import_n43: fix inconsistency between type of exception raised adn type of exception catched

mvrodriguez committed:
Translated using Weblate (Galician)

Currently translated at 100.0% (14 of 14 strings)

Translation: l10n-spain-12.0/l10n-spain-12.0-l10n_es_account_bank_statement_import_n43
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-12-0/l10n-spain-12-0-l10n_es_account_bank_statement_import_n43/gl/

jffernandez committed:
[FIX] l10n_es_account_bank_statement_import_n43: Change encoding for compatibility with BBVA files (#1056)

Some statement files include BOM data (https://es.wikipedia.org/wiki/Marca_de_orden_de_bytes) at the beginning of the file, making them non readable by current parser, as fixed positions are moved in these cases and thus expected data is not matched.

As ASCII, which is the encoding used in files according specification, is a subset of UTF-8, there's no danger in using it with the variant 'utf-8-sig' for handling if BOM marker is present or not.

chienandalu committed:
[FIX] l10n_es_account_bank_statement_import_n43: decoding import error

pedrobaeza committed:
[FIX] l10n_es_account_bank_statement_import_n43: Allow to parse on other encodings + flake8

pedrobaeza committed:
[IMP] l10n_es_account_bank_statement_import_n43: Full compatibility in tests with other statement import modules

Journal in context should be available since the beginning, as it can be used in default get for example.

pedrobaeza committed:
[FIX] l10n_es_account_bank_statement_import_n43: Don't fail on other formats

If other statement format is present, the file is properly decoded and stored
in `data_file` variable, but if the format is not N43, the loop will continue
and will try to decode the already decoded content in other encoding, giving an
error. We avoid that storing in a different variable the decode content.

etobella committed:
[IMP] l10n_es_account_bank_statement_import_n43: Set Date of the statment as the start date

Lopsanz committed:
[FIX] Match bank acount between file and journal

ValentinVinagre committed:
[IMP] l10n_es_account_bank_statement_import_n43: black, isort, prettier

ValentinVinagre committed:
[MIG] l10n_es_account_bank_statement_import_n43: Migration to 13.0

ValentinVinagre commented:
@HaraldPanten could you review please?

HaraldPanten reviewed (approved):
Revisión funcional 👍 

@misern2 @gelo-landoo @MiquelRForgeFlow éste os interesa 😉. ¿Podéis revisar?

AaronHForgeFlow reviewed (approved):
:+1:  LGTM

HaraldPanten commented:
@pedrobaeza ,

¿Cómo ves éste?¿Puedes echarle un ojo para ver si podemos fusionar?

¡Gracias! 👍

rafaelbn reviewed (approved):
👍

ValentinVinagre commented:
Hola @pedrobaeza ,
Creo que el módulo estaría listo para fusionar ¿Cómo lo ves?

pedrobaeza commented:
Tengo que mirar varios PRs pendientes en v12 primero, y ver si aplican aquí también.

ValentinVinagre commented:
> Tengo que mirar varios PRs pendientes en v12 primero, y ver si aplican aquí también.

Okay 👍

pedrobaeza commented:
Puedes echarles tú también un vistazo si quieres.

ValentinVinagre commented:
> Puedes echarles tú también un vistazo si quieres.

¿En un principio sería el https://github.com/OCA/l10n-spain/pull/1363 cierto?

pedrobaeza commented:
Y la issue #1350

ValentinVinagre commented:
> Y la issue #1350

Les echo un ojo. Merci.

pedrobaeza committed:
[FIX] l10n_es_account_bank_statement_import_n43: Compatible with other importers

- Don't store in `note` field temporary data for later operations
- Don't force N43 partner detection if other importers fill `note` field.
- Rename some methods for avoiding possible collapses.

Closes #1350 #1363

ValentinVinagre commented:
Hola @pedrobaeza ,
incorporo el fix d507cbd?

pedrobaeza commented:
Estoy yo ya con ello dando el último toque y voy a empujar en minutos la versió a fusionar.

ValentinVinagre commented:
> Estoy yo ya con ello dando el último toque y voy a empujar en minutos la versió a fusionar.

Perfecto entonces 👍

pedrobaeza reviewed (approved):
Incorporados últimos cambios, y repasada la migración. Listo para fusionar.

pedrobaeza commented:
/ocabot merge nobump

pedrobaeza commented:
Glitch de Travis:

/ocabot merge nobump
